src/dq_suite/output_transformations.py

```diff
@@ -571,3 +571,43 @@ def write_validation_result_tables(
             table_name=validation_table_name,
             schema=schema,
         )
+
+
+def get_highest_severity_from_validation_result(validation_result: dict, rules_dict: dict) -> str | None:
+    """
+    validation_result: dict containing ValidationResult["results"] (from checkpoint_result.run_results.values()[0])
+    rules_dict: Dictionary of rules containing rule_name and severity under the 'rules' key
+
+    Returns:
+        The highest severity level ('fatal', 'error', 'warning') or None
+    """
+
+    rules_by_name = {
+        rule["rule_name"]: rule["severity"]
+        for rule in rules_dict.get("rules", [])
+    }
+
+    failed_severities = []
+
+    severity_priority = {
+        "fatal": 3,
+        "error": 2,
+        "warning": 1,
+    }
+
+    for result in validation_result.get("results", []):
+        if result.get("success") is False:
+            expectation_type = result["expectation_config"]["type"]
+            expectation_type = expectation_type[len("expect_"):]
+            rule_name = "Expect" + "".join(
```

Comment:
I don't really understand why this (+2 lines above) is necessary? As far as I know, all the rule names should be `WrittenLikeThis` in the input JSON.

Reply:
Okay, understood. Interesting behaviour of GX then :) In that case using `humps` again seems to be the way indeed.